backends-common/elasticsearch-v7/src/main/java/org/apache/james/backends/es/v7/IndexCreationFactory.java

```diff
@@ -33,65 +36,121 @@
+            private final ImmutableList.Builder<AliasName> aliases;
+            private Optional<IndexCreationCustomElement> customAnalyzers;
+            private Optional<IndexCreationCustomElement> customTokenizers;
+
+            public Builder(int nbShards, int nbReplica, int waitForActiveShards, IndexName indexName) {
+                this.nbShards = nbShards;
+                this.nbReplica = nbReplica;
+                this.waitForActiveShards = waitForActiveShards;
+                this.indexName = indexName;
+                this.aliases = ImmutableList.builder();
+                this.customAnalyzers = Optional.empty();
+                this.customTokenizers = Optional.empty();
+            }
+
+            public Builder addAlias(AliasName aliasName) {
+                Preconditions.checkNotNull(aliasName);
+                this.aliases.add(aliasName);
+                return this;
+            }
+
+            public Builder customAnalyzers(IndexCreationCustomElement customAnalyzers) {
+                Preconditions.checkNotNull(customAnalyzers);
+                this.customAnalyzers = Optional.of(customAnalyzers);
+                return this;
+            }
+
+            public Builder customTokenizers(IndexCreationCustomElement customTokenizers) {
+                Preconditions.checkNotNull(customTokenizers);
+                this.customTokenizers = Optional.of(customTokenizers);
+                return this;
+            }
+
+            public IndexCreationPerformer build() {
+                return new IndexCreationPerformer(nbShards, nbReplica, waitForActiveShards, indexName, aliases.build(), customAnalyzers, customTokenizers);
+            }
+
+            public ReactorElasticSearchClient createIndexAndAliases(ReactorElasticSearchClient client) {
+                return build().createIndexAndAliases(client, Optional.empty());
+            }
+
+            public ReactorElasticSearchClient createIndexAndAliases(ReactorElasticSearchClient client, XContentBuilder mappingContent) {
+                return build().createIndexAndAliases(client, Optional.of(mappingContent));
+            }
+        }
+
         private final int nbShards;
         private final int nbReplica;
         private final int waitForActiveShards;
         private final IndexName indexName;
         private final ImmutableList<AliasName> aliases;
+        private final Optional<IndexCreationCustomElement> customAnalyzers;
+        private final Optional<IndexCreationCustomElement> customTokenizers;
 
         public IndexCreationPerformer(int nbShards, int nbReplica, int waitForActiveShards, IndexName indexName, ImmutableList<AliasName> aliases) {
+            this(nbShards, nbReplica, waitForActiveShards, indexName, aliases, Optional.empty(), Optional.empty());
+        }
+
+        public IndexCreationPerformer(int nbShards, int nbReplica, int waitForActiveShards, IndexName indexName, ImmutableList<AliasName> aliases,
```

Comment:
Shouldn't the constructors be private and we should only expose a static method returning a new Builder instance, forcing its use?